### PR TITLE
UIE-204 Narrow Ajax Usage pt13

### PR DESCRIPTION
**Pull request by msilva-broad**
- shifting ajax usage to directly use ajax sub-areas for some of the src/pages uses.
- improving mock types where possible.

### What
-

### Why
- improve code maintainability

### Testing strategy

- [ ]

### CODE REVIEW

File: src/pages/library/Code.test.js [hunk 1]
```diff
@@ -35,48 +37,14 @@ describe('Code page', () => {
...
-      },
-      expires_at: undefined,
-      state: undefined,
-      expires_in: 0,
-      expired: undefined,
-      scopes: [],
-      toStorageString: '',
-    };
-
-    Ajax.mockImplementation(() => {
```
Comment (msilva-broad) [inline src/pages/library/Code.test.js:60]: This must be a left-over from an earlier version of this test... Ajax.mockImplementation is called a 2nd time just after this, which negates this mock implementation completely.  So this block and its referenced bits like mockOidcUser can be removed.

File: src/pages/library/Datasets.test.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { dataCatalogStore } from 'src/libs/state';
 import { Datasets } from 'src/pages/library/Datasets';
 import { renderWithAppContexts as render } from 'src/testing/test-utils';
 
+jest.mock('src/libs/ajax/Metrics');
```
Comment (msilva-broad) [inline src/pages/library/Datasets.test.js:9]: mocking Metrics as a precaution

**Review by rjohanek — APPROVED** [review]
these testing changes look good to me, thanks!

**Comment by bahill** [comment]
Does someone need to review and accept the changes in Chromatic?
https://github.com/DataBiosphere/terra-ui/actions/runs/12035944664/job/33556055602?pr=5176#step:5:57

**Comment by bahill** [comment]
It also looks like there was a [parsing error with the mock_server](https://github.com/DataBiosphere/terra-ui/actions/runs/12035945291/job/33556057355?pr=5176#step:8:11) though the pact tests still passed.

**Review by bahill — COMMENTED** [review]
My questions above, I don't believe are blocking.
LGTM 👍

**Review by bahill — APPROVED** [review]

**Comment by msilva-broad** [comment]
> Does someone need to review and accept the changes in Chromatic? https://github.com/DataBiosphere/terra-ui/actions/runs/12035944664/job/33556055602?pr=5176#step:5:57

These PRs aren't making any function or visual changes, just test/type-safety cleanups.  The diffs you are seeing in Chormatic are because of us lagging in keeping up with accepting and advancing the baseline to diff from.  We were starting to ramp up Chormatic usage before the rift, but with Christina and many other devs gone, we are being lax about it as we regroup.

**Comment by msilva-broad** [comment]
> It also looks like there was a [parsing error with the mock_server](https://github.com/DataBiosphere/terra-ui/actions/runs/12035945291/job/33556057355?pr=5176#step:8:11) though the pact tests still passed.

pretty sure this is an existing issue, the PACT.io stuff is also in a holding pattern since the rift... open question as to how aggressively we move forward with contract testing in general.

**Pull request merged**